js/search/search.js

```diff
@@ -540,6 +544,34 @@ function clearSearchData() {
     }
 }
 
+/**
+ * Check if the file or folder exist or not
+ * @param type
+ * @returns {boolean}
+ */
+function isFileExist(type) {
+    let searchPath;
+
+    if (!this.userId) {
+        return false
+    }
+
+    switch (type) {
+        case 'USER_INDEX_PATH':
```

Comment:
Instead of using `switch case...` here, what we could do is keep all these values in an object and based on `type`'s value, we could return the corresponding value from the object.

Just a suggestion. That way, it's easier to grow your cases in the future rather than having long `switch case...` statements.